### PR TITLE
EVG-8252 Database error due to removed task queue

### PR DESCRIPTION
**Pull request by hhoke**
task queue clear after task queue removal in order to insert blank task queue and avoid errors

### CODE REVIEW

File: scheduler/wrapper.go [hunk 1]
```diff
@@ -53,13 +53,24 @@ func PlanDistro(ctx context.Context, conf Configuration, s *evergreen.Settings)
 			if err != nil {
 				grip.Error(err)
 			}
+			err = model.ClearTaskQueue(distro.Id)
```
Comment (bsamek) [inline scheduler/wrapper.go:56]: If you call `ClearTaskQueue`, is there a reason to also call `RemoveTaskQueues` first?

File: scheduler/wrapper.go [hunk 2]
```diff
@@ -53,13 +53,24 @@ func PlanDistro(ctx context.Context, conf Configuration, s *evergreen.Settings)
...
-				"err":       err,
-				"operation": "removed queue of disabled distro",
+				"distro":      distro.Id,
+				"err":         err,
+				"removed_len": queue_info.Length,
+				"operation":   "removed queue of disabled distro",
 			})
 		}
 
+		queue_info, _ = model.GetDistroQueueInfo(distro.Id)
```
Comment (bsamek) [inline scheduler/wrapper.go:68]: Is this debug logging?

File: scheduler/wrapper.go [hunk 3]
```diff
@@ -53,10 +53,15 @@ func PlanDistro(ctx context.Context, conf Configuration, s *evergreen.Settings)
...
 			}
+			err = model.ClearTaskQueue(distro.Id)
+			if err != nil {
+				grip.Error(err)
+			}
 			grip.Info(message.Fields{
-				"distro":    distro.Id,
-				"err":       err,
-				"operation": "removed queue of disabled distro",
+				"distro":      distro.Id,
```
Comment (ablack12) [inline scheduler/wrapper.go:61]: You can remove the "err" : err since you know error is nil

File: model/task_queue.go [hunk 4]
```diff
@@ -372,24 +382,54 @@ func updateTaskQueue(distro string, taskQueue []TaskQueueItem, distroQueueInfo D
...
 	})
+
+	catcher := grip.NewBasicCatcher()
+
+	distroQueueInfo, err := GetDistroQueueInfo(distroId)
+	if err != nil {
+		catcher.Add(errors.Wrap(err, "error clearing task queue"))
+	}
+	distroQueueInfo.Length = 0
+	distroQueueInfo.CountOverThreshold = 0
```
Comment (Kimchelly) [inline model/task_queue.go:398]: This should also clear expected duration and task group infos shouldn't it?

Comment (hhoke) [reply model/task_queue.go:398]: I wanted to err on the conservative side/ not blanking things but if you think that's advisable I have not specific objection,

Comment (Kimchelly) [reply model/task_queue.go:398]: I believe it should be fine to clear the task group info and will prevent some odd data inconsistencies in the task queue. I'm pretty sure it's used primarily for the host allocator to correctly figure out how many hosts have to be created (but that shouldn't matter if we're already clearing the task queue).

File: model/task_queue.go [hunk 5]
```diff
@@ -372,24 +382,61 @@ func updateTaskQueue(distro string, taskQueue []TaskQueueItem, distroQueueInfo D
...
+
+	err = clearTaskQueueCollection(distroId, TaskAliasQueuesCollection, distroQueueInfo)
+	if err != nil {
+		catcher.Add(errors.Wrap(err, "error clearing task queue"))
+	}
+	return catcher.Resolve()
+}
+
+// clearQueueInfo takes in a DistroQueueInfo and blanks out appropriate fields
+func clearQueueInfo(distroQueueInfo *DistroQueueInfo) {
```
Comment (Kimchelly) [inline model/task_queue.go:417]: Typically, it's not desirable to use in-out parameters as a way to produce output from go functions (unless it's absolutely necessary, like for `json.Unmarshal`) - it's fine to return a copied version of the struct though.

File: model/task_queue.go [hunk 6]
```diff
@@ -42,9 +42,19 @@ type DistroQueueInfo struct {
 }
 
 func GetDistroQueueInfo(distroID string) (DistroQueueInfo, error) {
+	rval, err := getDistroQueueInfoCollection(distroID, TaskQueuesCollection)
+	return rval, err
+}
+
+func GetDistroQueueInfoAlias(distroID string) (DistroQueueInfo, error) {
```
Comment (Kimchelly) [inline model/task_queue.go:49]: nit: can we call this `GetDistroAliasQueueInfo`?

File: model/task_queue.go [hunk 7]
```diff
@@ -42,9 +42,19 @@ type DistroQueueInfo struct {
...
+	rval, err := getDistroQueueInfoCollection(distroID, TaskQueuesCollection)
```
Comment (Kimchelly) [inline model/task_queue.go:50]: Should `TaskQueuesCollection` be `TaskAliasQueuesCollection`?

File: model/task_queue.go [hunk 8]
```diff
@@ -372,24 +382,61 @@ func updateTaskQueue(distro string, taskQueue []TaskQueueItem, distroQueueInfo D
 	return errors.WithStack(err)
 }
 
-func ClearTaskQueue(distro string) error {
+func ClearTaskQueue(distroId string) error {
```
Comment (Kimchelly) [inline model/task_queue.go:385]: Can we get a doc comment for the difference between `ClearTaskQueue` and `RemoveTaskQueue`?

File: model/task_queue.go (hunk 7 above)
Comment (hhoke) [reply model/task_queue.go:50]:
```go
// ClearTaskQueue removes all tasks from task queue by updating them with a blank queue,
//and modifies the queue info.
//This is in contrast to RemoveTaskQueues, which simply deletes these documents.
```

Comment (Kimchelly) [reply model/task_queue.go:50]: LGTM, mod linting nit - adding spacing between the `//` and comment text.

**Pull request merged**